### PR TITLE
fix(dom.js): added support for nextjs

### LINKED ISSUE #80: Getting "document is not defined" with Next.js
Hi there,

I was trying your awesome library that works seamlessly without any framework. I tried it with Next.js and no matter what I do, I am constantly getting the error as attached "document is not defined". Any idea on how to solve this issue and use it with next.js.

splitting error

### PR DESCRIPTION
**Pull request by EmmanuelOmoiya**
Added support for nextjs by fixing error "document is not defined" when used in a nextjs application
Also added build support for ES6+ using terser
fix #80

### CODE REVIEW

**Comment by EmmanuelOmoiya** [comment]
Kindly review @shshaw & @notoriousb1t .

File: src/utils/dom.js [hunk 1]
```diff
@@ -61,5 +61,5 @@ export function $(e, parent) {
             ? // a single element is wrapped in an array
               [e]
             : // selector and NodeList are converted to Element[]
-              [].slice.call(e[0].nodeName ? e : (parent || root).querySelectorAll(e));
+              [].slice.call(e[0].nodeName ? e : document.querySelectorAll(e));
```
Comment (shshaw) [inline src/utils/dom.js:64]: How does this fix "Document not defined?" This is also a departure from the expected logic that finds the elements inside a parent if one is provided.

Comment (shshaw) [reply src/utils/dom.js:64]: As far as I can tell, this is the only logic change. If there's something I missed, do let me know. In the future, try to scope PRs to more focused changes so real changes are more apparent.

**Comment by EmmanuelOmoiya** [comment]
Kindly resolve the "Document not defined" error as it's a big issue with using your package with nextjs.

Works on my end as implemented in my latest commit.

**Pull request closed without merging**